openml/flows/flow.py

```diff
@@ -1,80 +1,358 @@
 from collections import OrderedDict
+import re
+
+import six
 import xmltodict
-import sklearn
 
 from .._api_calls import _perform_api_call
-from .functions import _check_flow_exists
+from openml.util import oml_cusual_string
 
 
 class OpenMLFlow(object):
     """OpenML Flow. Stores machine learning models.
 
+    Flows should not be generated manually, but by the function
+    :meth:`openml.flows.create_flow_from_model`. Using this helper function
+    ensures that all relevant fields are filled in.
+
     Parameters
     ----------
-    model : scikit-learn compatible model
-        The model the flow consists of. The model needs to have fit and predict methods.
-    description : string
+    name : str
+        Name of the flow. Is used together with the attribute `external_version`
+        as a unique identifier of the flow.
+    description : str
         Description of the flow (free text).
-    creator : string
-        FIXME
-    contributor : string
-        FIXME
-    tag : string
-        FIXME
+    model : object
+        ML model which is described by this flow.
+    components : OrderedDict
+        Mapping from component identifier to an OpenMLFlow object.
+    parameters : OrderedDict
+        Mapping from parameter name to the parameter default value. The
+        parameter default value must be of type `str`, so that the respective
+        toolbox plugin can take care of casting the parameter default value to
+        the correct type.
+    parameters_meta_info : OrderedDict
+        Mapping from parameter name to `dict`. Stores additional information for
+        each parameter. Required keys are `data_type` and `description`.
+    external_version : str
+        Version number of the software the flow is implemented in. Is used
+        together with the attribute `name` as a uniquer identifier of the flow.
+    uploader : str
+        OpenML user ID of the uploader. Filled in by the server.
+    tags : list
+        List of tags. Created on the server by other API calls.
+    binary_url : str
+        ??? - don't use - implemented because it is used by flows on the server
+    binary_format : str
+        ??? - don't use - implemented because it is used by flows on the server
+    binary_md5 : str
+        ??? - don't use - implemented because it is used by flows on the server
+    version : str
+        OpenML version of the flow.
```

Comment:
Makes sense. I'm also not sure about these attributes so I will copy the description from the xsd schema and add a sentence why one should not use those fields.